programs/create-genesis/ee_genesis/ee_genesis_builder.cpp

```diff
@@ -409,7 +408,78 @@ void genesis_ee_builder::build_pinblocks() {
     }
 }
 
-void genesis_ee_builder::build(const bfs::path& out_dir) {
+void ee_genesis_builder::build_usernames() {
+    const auto& app = _info.golos.names.issuer;
+
+    out_.usernames.start_section(config::system_account_name, N(domain), "domain_info");
+
+    out_.usernames.insert(mvo
+        ("owner", app)
+        ("linked_to", app)
+        ("name", _info.golos.domain)
+    );
+
+    out_.usernames.start_section(config::system_account_name, N(username), "username_info");
+
+    for (const auto& auth : _state.auths) {
+        const auto& name = auth.account.str(_state.accs_map);
+        const auto& hash = generate_name(name);
+        out_.usernames.insert(mvo
+            ("creator", app)
+            ("owner", hash)
+            ("name", name)
+        );
+    }
+}
+
+void ee_genesis_builder::build_balances() {
```

Comment:
Logic duplicates with genesis_create.cpp:store_balances (https://github.com/GolosChain/cyberway/pull/714/files#diff-abadb396f94149e5a62c6952dbdaf5daR578)?

How will we guarantee the consistency of these functions?

Reply:
@s-medvedev It is better to move duplicating logic to state_reader. Will fix in another PR

Reply:
building structures to be stored is not work for the state_reader.
it's better to ask genesis_create to export some objects it creates, so ee_builder can reuse them without recalculating balances, etc